### PR TITLE
Lookup service testing

### PR DESCRIPTION
**Pull request by lwander**
@duftler or @jtk54 or @danielpeach PTAL

### CODE REVIEW

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/config/v1/HalconfigParser.java [hunk 1]
```diff
@@ -74,41 +74,49 @@ Halconfig parseConfig(InputStream is) throws UnrecognizedPropertyException {
...
     if (!reload && halconfig != null) {
-      halconfig.parentify();
-      return halconfig;
+      res = halconfig;
+    } else {
+      try {
+        InputStream is = getHalconfigStream();
+        res = parseHalconfig(is);
+      } catch (FileNotFoundException e) {
+      } catch (UnrecognizedPropertyException e) {
```
Comment (duftler) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/config/v1/HalconfigParser.java]: `catch (UnrecognizedPropertyException | ParserException | ScannerException)`

File: halyard-config/src/test/groovy/com/netflix/spinnaker/halyard/config/model/v1/NodeSpec.groovy [hunk 2]
```diff
@@ -122,4 +124,24 @@ class NodeSpec extends Specification {
...
+    def iterator = node.getChildren()
+
+    when:
+    def child = iterator.getNext()
+    def i = 0
+
+    then:
+    child != null
+    while (child != null) {
+      i += 1
```
Comment (duftler) [inline halyard-config/src/test/groovy/com/netflix/spinnaker/halyard/config/model/v1/NodeSpec.groovy]: `i++`?

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/config/v1/HalconfigParser.java (hunk 1 above)
Comment (lwander) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/config/v1/HalconfigParser.java]: The `ParseConfigExceptions` dispatch constructors based on the exception type - and the `Exception2 | Exception2` squashes the exception type to their common super-class.

Comment (duftler) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/config/v1/HalconfigParser.java]: ah, I recall us having a similar conversation earlier too

**Pull request merged**